File: sample-services/npm/src/commands/update-something.ts
import { ICommand } from "../abstractions/command";
import { CommandHandlerBase } from "../abstractions/command-handler";

export interface IUpdateSomethingCommand extends ICommand {
    someId: number;
    someNewData: string;
}

export class UpdateSomethingCommand implements IUpdateSomethingCommand {
    static readonly TYPE = "SampleServices.Commands.UpdateSomethingCommand";
    readonly type: string;
    someId: number;
    someNewData: string;
    constructor(state?: Partial<IUpdateSomethingCommand>) {
        this.someId = 0;
        this.someNewData = "default";
        state && Object.assign(this, JSON.parse(JSON.stringify(state)));
        this.type = UpdateSomethingCommand.TYPE;
    }
}

export interface IUpdateSomethingHandler extends CommandHandlerBase<IUpdateSomethingCommand, number> { }

File: sample-services/dotnet/src/commands/update-something.cs
using Sample.Services.Abstractions;

namespace Sample.Services.Commands
{
    public class UpdateSomethingCommand : CommandBase, ICommand
    {
        public const string TYPE = nameof(UpdateSomethingCommand);
        public override string Type => TYPE;
        public int SomeId { get; set; }
        public string SomeNewData { get; set; }
        public UpdateSomethingCommand()
        {
            SomeId = 0;
            SomeNewData = "default";
        }
    }
    
    public interface IUpdateSomethingHandler : ICommandHandler<UpdateSomethingCommand, int> { }
}